src/components/form/FormFieldCompanyAddress.vue

```diff
@@ -34,86 +34,70 @@ import DialogDefault from 'src/components/global/DialogDefault.vue';
 import { useValidation } from 'src/composables/useValidation';
 
 // types
-import type {
-  FormCompanyAddressFields,
-  FormOption,
-} from 'src/components/types/Form';
+import type { FormOption } from 'src/components/types/Form';
 
 export default defineComponent({
   name: 'FormFieldCompanyAddress',
   components: {
     DialogDefault,
   },
   props: {
-    BusinessId: {
-      type: String,
+    options: {
+      type: Array as () => FormOption[],
+      required: true,
+    },
+    modelValue: {
+      type: String as () => string | null,
+      required: true,
     },
   },
-  emits: ['update:formValue'],
+  emits: ['update:modelValue'],
   setup(props, { emit }) {
-    const address = ref<string | null>(null);
-    const isDialogOpen = ref<boolean>(false);
-
-    const options: FormOption[] = [
-      {
-        label: 'Address 1',
-        value: 'address-1',
-      },
-      {
-        label: 'Address 2',
-        value: 'address-2',
-      },
-    ];
+    const address = computed<string | null>({
+      get: () => props.modelValue,
+      set: (value: string | null) => emit('update:modelValue', value),
+    });
 
-    // Lookup dictionary which provides the full values based on address ID.
-    // TODO: validate the possibility of using this method
-    const addresses: { [key: string]: FormCompanyAddressFields } = {
-      'address-1': {
-        street: 'Street 1',
-        streetNumber: '123',
-        city: 'City 1',
-        zip: '1234',
-        referenceCity: 'Ref City 1',
-        department: 'Department 1',
+    watch(
+      () => props.options,
+      () => {
+        address.value = null;
       },
-      'address-2': {
-        street: 'Street 2',
-        streetNumber: '123',
-        city: 'City 2',
-        zip: '1234',
-        referenceCity: 'Ref City 2',
-        department: 'Department 2',
-      },
-    };
-
-    const onUpdate = (): void => {
-      // wait for next tick to emit the value after update
-      nextTick((): void => {
-        if (address.value) {
-          emit('update:formValue', addresses[address.value as string]);
-        }
-      });
-    };
+    );
 
-    const onClose = (): void => {
-      isDialogOpen.value = false;
-    };
+    const { isFilled } = useValidation();
+    const { isDialogOpen, onClose } = useDialog();
 
     const onSubmit = async (): Promise<void> => {
-      // noop
+      // TODO: Add address via API
       isDialogOpen.value = false;
     };
 
-    const { isFilled } = useValidation();
+    /**
+     * Provides dialog behaviour
+     * @returns
+     * - `isDialogOpen` (boolean): Whether the dialog is open.
+     * - `onClose` (function): Runs when the dialog is closed.
+     */
+    function useDialog() {
```

Comment:
Really you need standalone `useDialog()` func wrapper here for returning consts? Make it simple as possible, please without wrapper func.

```ts
const isDialogOpen = ref<boolean>(false);

const onClose = (): void => {
  isDialogOpen.value = false;
};
```

Reply:
Or you want reuse this function in another component?

Reply:
It was more an issue of separation of concerns: individual behaviours are encapsulated into composables (different file) or inline composables (same file) - [example by Evan You](https://gist.github.com/yyx990803/8854f8f6a97631576c14b63c8acd8f2e)
But yes, in this case the composable can be shared with `FormFieldCompany` and `FormFieldSelectTable`.

Reply:
Ok.